api/app/change_log/tracker.py

```diff
@@ -0,0 +1,31 @@
+from copy import deepcopy
+
+
+class BaseChangeTracker:
+    def __init__(self, instance):
+        self.instance = instance
+        self.data = {}
+
+    def store_state(self):
+        for field in self.instance._meta.fields:
+            self.data[field.name] = deepcopy(getattr(self.instance, field.name))
+
+    @property
+    def instance_changed(self):
+        for field in self.instance._meta.fields:
+            if getattr(self.instance, field.name) != self.data.get(field.name):
+                return True
+        return False
+
+    def changed_data(self):
```

Comment:
See comment on `ChangeLogger`. Output of this function is not necessarily save to store in a `JSONField` (as used by the `Log` model.

Reply:
True